subsys/bluetooth/host/gatt.c

```diff
@@ -878,6 +878,8 @@ static void db_hash_process(struct k_work *work)
 {
 	if (!atomic_test_bit(gatt_sc.flags, DB_HASH_VALID)) {
 		db_hash_gen();
+	} else {
+		return;
```

Comment:
If we generate the hash before settings are loaded (e.g. if a GATT service is registered before settings are loaded), we could accidentally skip setting `DB_HASH_LOAD_PROC` and performing the related operations needed after db_hash load from settings is completed.

Reply:
btw. Shouldn't the condition below be `if (already_processed)` (without the negation) ?

Reply:
ugh yeah. shouldn't have refactored this I guess